Development environment/Laravel Homestead setup.md

```diff
@@ -0,0 +1,147 @@
+
+```yml
+folders:
+    # Local path to the directory in which we cloned Homestead repo in
+    - map: ~/homestead
+      # Path to the directory from the virtual machine
+      to: /home/vagrant/homestead
+```
+
+For multiple sites, each with its own instance:
+
+```yml
+folders:
+    - map: ~/homestead/project-1
+      to: /home/vagrant/homestead/project-1
+
+    - map: ~/homestead/project-2
+      to: /home/vagrant/homestead/project-2
+```
+
+Other important thing to configure is sites. They are stored in one or many shared folders and they are defined with domain name that will map to a site folder:
+
+```yml
+sites:
+    # Local domain name
+    - map: homestead.test
+      # Location of the sites public folder
+      to: /home/vagrant/homestead/www/
+  ```
+
+Sites domain name must also be mapped to the actual IP address of the Vagrant server. That is done by editing `hosts` file of the local machine. This can be done manually, but for a more automated approach it is recommended to use [vagrant-hostsupdater](https://github.com/cogitatio/vagrant-hostsupdater) plugin.
+
+Install it by running: `vagrant plugin install vagrant-hostsupdater` inside the terminal while in `/homestead` directory. Homestead already has config for `vagrant-hostsupdater` plugin and should map all sites domains to Vagrants IP address.
+By default IP address is defined in `Homestead.yml` but you can change it to some other address.
+
+Note for Windows users: You'll need to change permissions on `hosts` file to allow current user write access to that file.
+
+Last thing for basic configuration is database property. With this we define all databases that will be created on server startup or reprovision.
+```yml
+databases:
+  - homestead
+  - wordpress
+  ```
+
+Other thing to note is that default Homestead database name is `homestead` and password is `secret`.
+
+## Using Homestead
+
+When we are done with Homestead configuration, we can start the server with command `vagrant up`, stop the server with `vagrant halt`, restart it with `vagrant reload`. If something is changed in `Homestead.yml` configuration server will need to be reprovisioned so that all changes in configuration are reflected on our sites. To reprovision a server use `vagrant reload --provision`.
+
+Now you have everything to run your WordPress site. Download the instance of WordPress in your preferred way and set it up in `wp-config.php`. To finish installation go to your local site URL (eg. `wordpress.test`).
+
+## Extra features
+
+### Mailhog
+
+Mailhog is really useful tool for intercepting incoming and outgoing emails on local machine. To set it up first create `.env` file in `/homestead` directory and place Mailhog config inside:
+```
+MAIL_DRIVER=smtp
+MAIL_HOST=localhost
+MAIL_PORT=1025
+MAIL_USERNAME=null
+MAIL_PASSWORD=null
+MAIL_ENCRYPTION=null
+```
+After that Mailhog interface is accessible on `http://localhost:8025`
+
+### Accessing Vagrant from inside
```

Comment:
Not specific to Homestead but maybe you should add a part about MySQL and make it stand out. ("if you need to access your database, you need to do it from inside").

For me that one took the longest to "click" about developing using Vagrant.

Reply:
👍